draft-ietf-protected-audience-key-value-services.md

```diff
@@ -0,0 +1,518 @@
+`acceptCompression` (see {{request-schema}}) field in the request and the server behavior.
+
+## Core Request Data {#request}
+
+Requests are not compressed and have a tree-like hierarchy:
+
+-   Each request contains one or more partitions. Each partition is a collection of keys that can be
+    processed together by the service without any potential privacy leakage. Keys from one interest
+    group must be in the same partition. Keys from different interest groups with the same joining
+    site (see https://wicg.github.io/turtledove/#joining-interest-groups) may or may not be in the same partition.
+-   Each partition contains one or more key groups. Each key group has its unique attributes among
+    all key groups in the partition. The attributes are represented by a list of `tags`. Besides
+    tags, the key group contains a list of keys to look up.
+-   Each partition has a unique id.
+-   Each partition has a compression group field. Results of partitions belonging to the same
+    compression group can be compressed together in the response. Different compression groups must
+    be compressed separately (see {{compression-group}}).
+
+### Encryption {#request-encryption}
+
+The request is encrypted with [HPKE]. The request uses a repurposed [OHTTP] encapsulation format
+(see {{Section 4.3 of OHTTP}} and {{Section 4.6 of OHTTP}}) with a
+`message/ad-auction-trusted-signals-request` media type instead of `message/bhttp request`.
+
+### Request Schema {#request-schema}
+
+The request is a [CBOR] encoded message with the following [CDDL] schema:
+
+~~~~~ cddl
+request = {
+    ? acceptCompression: [* tstr],
+    ; A list of supported response compression algorithms; must contain at least one of "none", "gzip", "brotli"
+    partitions: [* partition],
+    ; A list of partitions. Each must be processed independently. Accessible by UDF.
+}
+
+partition = {
+    id: uint,
+    ; Unique id of the partition in this request. Used by responses to refer to request partitions.
+    compressionGroupId: uint,
+    ; Unique id of a compression group in this request. Only partitions belonging to the same compression group will be compressed together in the response
+    ? metadata: partitionMetadata,
+    arguments: [* requestArgument],
+    ; One group of keys and common attributes about them
+}
+;Single partition object. A collection of keys that can be processed together.
+
+
+partitionMetadata = {
+    ? hostname: tstr,
+    ; The hostname of the top-level frame calling runAdAuction
+    ? experimentGroupId: tstr,
+    ? slotSize: tstr,
+    ; Available if trustedBiddingSignalsSlotSizeMode=slot-size. In the form of <width>,<height>
+    ? allSlotsRequestedSizes: tstr,
+    ; Available if trustedBiddingSignalsSlotSizeMode=all-slots-requested-sizes. In the form of <width1>,<height1>,<width2>,<height2>,...
+}
+
+requestArgument = {
+    ? tags: [* tstr],
+    ; List of tags describing this group's attributes
```

Comment:
Are these picked from the below list of "Available Tags"?  If so, we should link to that section.
If this is an enumeration, we should declare its possible values rather than use the type "tstr" I think like Russ mentioned above for `acceptCompression`

Reply:
@peiwenhu are we prescribing that it is from the list of available tags? I don't think we should, since this will be input to the UDF.

Reply:
Discussed offline, we don't really want to restrict it to only those tags.